cmd/gossamer/main.go

```diff
@@ -397,13 +397,11 @@ func buildSpecAction(ctx *cli.Context) error {
 	}
 
 	if outputPath := ctx.String(OutputSpecFlag.Name); outputPath != "" {
-		if err = dot.WriteGenesisSpecFile(res, outputPath); err != nil {
-			return err
-		}
-	} else {
-		fmt.Printf("%s", res)
+		return dot.WriteGenesisSpecFile(res, outputPath)
```

Comment:
nit it would be nice to wrap the error since the function is rather large

```suggestion
		err = dot.WriteGenesisSpecFile(res, outputPath)
		if err != nil {
			return fmt.Errorf("cannot write genesis spec file: %w", err)
		}
```